nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -275,15 +290,14 @@ def forward(
             encoder_input = None
 
         # Call forward on GPT model with preprocessed embeddings
-        if self.autocast_dtype == torch.float32:
+        if self.frozen_model.mcore_gpt:
             output = self.frozen_model.model(
                 input_ids=None,
                 position_ids=None,
-                encoder_input=encoder_input,
+                decoder_input=encoder_input,
                 attention_mask=attention_mask,
                 labels=labels,
-                set_inference_key_value_memory=set_inference_key_value_memory,
-                inference_max_sequence_len=inference_max_sequence_len,
+                inference_params=inference_params,
```

Comment:
outside this PR, since inference_params were added, we'd want to remove the other inference related keys and just use the config